src/cli/cmd/init/mod.rs

```diff
@@ -0,0 +1,347 @@
+mod dev_shell;
+mod project;
+mod prompt;
+mod template;
+
+use clap::Parser;
+use color_eyre::eyre::Result;
+use project::Project;
+use prompt::Prompt;
+use std::collections::HashMap;
+use std::fs::File;
+use std::io::Write;
+use std::path::PathBuf;
+use std::process::ExitCode;
+use url::Url;
+
+use super::FlakeHubClient;
+
+use self::{dev_shell::DevShell, prompt::MultiSelectOption, template::TemplateData};
+
+use super::{CommandExecute, FhError};
+
+const CARGO_TOOLS: &[&str] = &[
+    "audit", "bloat", "cross", "edit", "outdated", "udeps", "watch",
+];
+const NODE_VERSIONS: &[&str] = &["18", "16", "14"];
+const PYTHON_VERSIONS: &[&str] = &["3.11", "3.10", "3.09"];
+const PYTHON_TOOLS: &[&str] = &["pip", "virtualenv", "pipenv"];
+const RUBY_VERSIONS: &[&str] = &["3.2", "3.1"];
+const GO_VERSIONS: &[&str] = &["20", "19", "18", "17"];
+const COMMON_TOOLS: &[&str] = &["curl", "Git", "jq", "wget"];
+const SYSTEMS: &[MultiSelectOption] = &[
+    MultiSelectOption(
+        "x86_64-linux",
+        "Linux on a 64-bit x86 processor, like Intel or AMD",
+    ),
+    MultiSelectOption("aarch64-linux", "Linux on a 64-bit Arm processor"),
+    MultiSelectOption("x86_64-darwin", "macOS on Intel CPUs"),
+    MultiSelectOption(
+        "aarch64-darwin",
+        "macOS on Apple Silicon, like the M1 or M2 chips",
+    ),
+];
+const PHP_VERSIONS: &[&str] = &["8.3", "8.2", "8.1", "8.0", "7.4", "7.3"];
+const JAVA_VERSIONS: &[&str] = &["19", "18", "17", "16", "15"];
+
+// Helper functions
+fn version_as_attr(v: &str) -> String {
+    v.replace('.', "")
+}
+
+/// Create a new flake.nix using an opinionated interactive initializer.
+#[derive(Parser)]
+pub(crate) struct InitSubcommand {
+    #[clap(long, short, default_value = ".")]
+    root: PathBuf,
+
+    #[clap(long, short, default_value = "./flake.nix")]
+    output: PathBuf,
+
+    #[clap(from_global)]
+    api_addr: url::Url,
+}
+
+#[async_trait::async_trait]
+impl CommandExecute for InitSubcommand {
+    async fn execute(self) -> Result<ExitCode> {
+        let mut inputs: HashMap<String, String> = HashMap::new();
+        let mut dev_shell_packages: Vec<String> = Vec::new();
+        let mut dev_shells: HashMap<String, DevShell> = HashMap::new();
+        let mut overlay_refs: Vec<String> = Vec::new();
+        let mut overlay_attrs: HashMap<String, String> = HashMap::new();
+        let mut env_vars: HashMap<String, String> = HashMap::new();
+
+        if self.output.exists() && !Prompt::bool("A flake.nix already exists in the current directory. Would you like to overwrite it?")? {
+            println!("Exiting. Let's a build a new flake soon, though :)");
+            return Ok(ExitCode::SUCCESS);
+        }
+
+        println!("Let's build a Nix flake!");
+
+        let project = Project::new(self.root);
+        let description = Prompt::maybe_string("An optional description for your flake")?;
```

Comment:
Can we get all these `Prompt`s colored? I think we've used https://docs.rs/owo-colors/3.5.0/owo_colors/ in the past  to great effect.

And/or suffix it with a colon:

```suggestion
        let description = Prompt::maybe_string("An optional description for your flake:")?;
```

because right now it looks like it's all "one thing": 
![image](https://github.com/DeterminateSystems/fh/assets/28582702/f197b5fd-aca1-41ab-b579-c875117855c3)

Reply:
I just noticed that it colors the user's input _after_ they've confirmed it:

![image](https://github.com/DeterminateSystems/fh/assets/28582702/6153a619-74de-41ec-b698-19ae0f622cdd)

If possible, it would be nice if we could swap the functionality so that the prompt is colored (always), and the user input is unstyled (so they both stand apart from each other all the time)